### PR TITLE
fix(es/minifier): inline vars declared in cond

### LINKED ISSUE #5367: Cannot inline variable declared in condition
### Describe the bug

They will be treated as `cond_init` while they're not.

### Input code

```typescript
export function foo(flag) {
    if (flag) {
        var a = 10
        console.log(a)
    }
}
```
or
```js
export function foo(flag) {
    if (flag) return
    var a = 10
    console.log(a)
}
```

### Config

_No response_

### Playground link

https://play.swc.rs/?version=1.2.222&code=H4sIAAAAAAAAA0utKMgvKlFIK81LLsnMz1NIy8%2FXSMtJTNdUqOZSAILMNAUUPgiUJRYpJCrYKhgawIWS8%2FOK83NS9XLy0zUSNcHCtVy1ANwgUt1dAAAA&config=H4sIAAAAAAAAA22TQZLjIAxF78I6i6ksepEDzG7OQCkgHGZs5EIiHVcqdx%2FhJBAnvTPPX0L6Elfzl505XM0MmTHXL16SwMUcDLoJ2OU4i9kZj44yCGU2B8kFdxqoogAj421nBPKAUoN4%2F2u%2F14CRiPEh2JkpphiWmt7RNGdkrt8aVCZMwk0HOdN3u%2BJINCKkj7MFtjEJDph7qKNxhJnRnqEXWW%2BDHJnSBhVBb%2BdM8wtNPkqkpMmezCN468hjAzGjk3jGLtEcKkmshbVKVujxWIahWnpX4hnGAtJy4WW1QW9scSeKLDaUxG9oU%2Bkd3Zt8qGKwGaXk1GZDMW18%2BIeodY7AnGDCHrnyoHPYKsMPqpiCDlGWRnXCvaqEgzZnYwytw1o1ZondrYy%2BOKwdup78ATcNcfRoMQR1uwXzdxR36slkmZFCO6prEPoM7kf73KctrSvwAX9rVfI6jgefQE7vjJfpSONHignlRP4DqxFC7zDr9l7md1qSR7UZ%2FcuPwvXY1lSHL2TH%2BtqePutCaIwdRjq29dVHOUEa2lreKiBfKriu5q2v9cvc%2But8rhP%2FeQjXRP8B2jXHlCQEAAA%3D

### Expected behavior

They can be inlined

### Actual behavior

_No response_

### Version

1.2.222

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
closes #5367

### CODE REVIEW

File: crates/swc_ecma_minifier/tests/fixture/next/feedback-util-promisify/chunks/pages/_app-72ad41192608e93a/output.js [hunk 1]
```diff
@@ -2229,20 +2229,6 @@
                                     configurable: !0
                                 }), t;
                             }
```
Comment (Austaras) [inline crates/swc_ecma_minifier/tests/fixture/next/feedback-util-promisify/chunks/pages/_app-72ad41192608e93a/output.js:2231]: Why do people write code like this...

Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/next/feedback-util-promisify/chunks/pages/_app-72ad41192608e93a/output.js:2231]: It works ™

**Comment by Austaras** [comment]
This would disable optimization for
```js
var a = 123

function a() {
  //
}
```
But luckily people don't write code like this, or at least there's none in the tests.

**Comment by Austaras** [comment]
Maybe in future we could optimize
```js
if (a) {
  var b = 1
} else {
  var b = 2
}
```
But the benefit of this is decling daily as people are shipping es6.

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_minifier

**Review by kdy1 — APPROVED** [review]

**Pull request merged**